examples/object_detection_2d/object_detection_2d/seed_test_suite.py

```diff
@@ -134,14 +134,53 @@ def seed_test_suite_by_brightness(test_suite_name: str, complete_test_case: Test
     print(f"created test suite '{test_suite.name}' v{test_suite.version}")
 
 
+def seed_test_suite_by_bounding_box_size(test_suite_name: str, complete_test_case: TestCase) -> None:
+    stratification_logic_map = {
+        "small": lambda area: area < 10_000,
+        "medium": lambda area: 10_000 <= area < 60_000,
+        "large": lambda area: 60_000 <= area,
+    }
+
+    def filter_gt_bboxes(gt: GroundTruth, filter_fn: Callable[[float], bool]) -> GroundTruth:
+        bboxes, ignored_bboxes = [], gt.ignored_bboxes
+        for bbox in gt.bboxes:
+            bboxes.append(bbox) if filter_fn(bbox.area()) else ignored_bboxes.append(bbox)
+        return GroundTruth(bboxes=bboxes, ignored_bboxes=ignored_bboxes)
+
+    # create each test case by stratification
+    test_cases: List[TestCase] = []
+    for name, fn in stratification_logic_map.items():
+        samples_with_filtered_bboxes = []
+        for ts, gt in complete_test_case.iter_test_samples():
+            filtered_ground_truth = filter_gt_bboxes(gt, fn)
+            if filtered_ground_truth.n_bboxes > 0:
+                samples_with_filtered_bboxes.append((ts, filtered_ground_truth))
+        new_test_case = TestCase(
+            f"bounding box size :: {name} :: {DATASET}",
+            test_samples=samples_with_filtered_bboxes,
+            reset=True,
+        )
+        test_cases.append(new_test_case)
+
+    # create the test suite with the complete test case and new test cases
+    test_suite = TestSuite(
+        test_suite_name,
+        description=f"{TEST_SUITE_DESCRIPTION}, stratified by `small`, `medium`, and `large` sized bounding boxes",
+        test_cases=[complete_test_case, *test_cases],
+        reset=True,
+    )
+    print(f"created test suite '{test_suite.name}' v{test_suite.version}")
+
+
 def main(args: Namespace) -> None:
     kolena.initialize(os.environ["KOLENA_TOKEN"], verbose=True)
 
     complete_test_case = create_complete_transportation_case(args)
 
     # organize test suite names with its generator
     test_suites: List[Tuple[str, Callable[[str, TestCase], TestSuite]]] = [
-        (f"{DATASET} :: transportation brightness [Object Detection]", seed_test_suite_by_brightness),
+        (f"{DATASET} :: transportation by brightness [Object Detection]", seed_test_suite_by_brightness),
+        (f"{DATASET} :: transportation by bounding box size [Object Detection]", seed_test_suite_by_bounding_box_size),
```

Comment:
Running through this on our trunk stack right now -- hopefully a useful example coming up soon.